### ISSUE TITLE
https://github.com/terraform-providers/terraform-provider-aws/issues/8447

### ISSUE THREAD

**Issue by toredash** [terraform-providers/terraform-provider-aws#11324]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

rds_cluster_instance should support specifying if its a writer or reader.

AFAIK, it is random what instance becomes the writer when creating a new DB cluster. By adding this support, one can decide on initial creation what instance become the writer, or promote another instance to become the writer instance.

There can only be one writer within one aws_rds_cluster / cluster_identifier

### New or Affected Resource(s)

rds_cluster_instance

### Potential Terraform Configuration

AWS provider already exposes if an rds instance is a writer or not:
```
# terraform state show aws_rds_cluster_instance.cluster_instance_primary
resource "aws_rds_cluster_instance" "cluster_instance_primary" {
[...]
    writer                       = false
[...]
}
```

One should be able to specify at most 1 `aws_rds_cluster_instance` to be `writer =  true`. API from AWS to support this function is documented here:
https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_FailoverDBCluster.html

Example code for how the TF resource could look like.
```hcl
resource "aws_rds_cluster_instance" "cluster_instance_primary" {
  identifier         = "writer-primary"
  cluster_identifier = module.rds.cluster_id
  instance_class     = "db.t3.small"

  writer = true

  db_parameter_group_name = aws_db_parameter_group.matomo-parameter-group.name
  db_subnet_group_name    = module.rds.db_subnet_group_name
}
```

TF should fail if more than one aws_rds_cluster_instance within the same cluster_identifier specifies `writer = true`.

### References

https://github.com/terraform-providers/terraform-provider-aws/issues/8447

**Comment by nirsv** [comment]
Any updates on this?
I also want to add read replica for my rds cluster (aurora postgresql).
The resource aws_rds_cluster_instance doesn’t have such option although in the AWS console I do have such option.
What am I missing?

**Comment by hongkongkiwi** [comment]
How to escalate this? it's been 10 months, this is obviously a small omission but important for finer control over instances.

**Comment by sunilkumarvytla** [comment]
Any update on this?

**Comment by sunilkumarvytla** [comment]
> There is a workaround. You can take advantage of the fact that the first instance created in a cluster will be the writer, and subsequent instance(s) created will be read replica(s). Give the desired writer instance its own resource definition, and in the read replica instance resource definition(s) use `depends_on` to establish an explicit dependency on the writer instance. This will cause Terraform to create the writer instance first, making it the writer, and all others read replicas.


@cbatson Could you please be more precise ? 

Can I use like the below?

`resource "aws_rds_cluster_instance" "main_reader" {
  identifier         = "${var.name}-reader"
  depends_on = [aws_rds_cluster_instance.main_primary.id]`

**Comment by cbatson** [comment]
@sunilkumarvytla Yes, that is the idea. Except, for `depends_on`, specify the resource itself, not the `id` attribute. As in,

```
resource "aws_rds_cluster_instance" "main_reader" {
   identifier         = "${var.name}-reader"
   depends_on    = [aws_rds_cluster_instance.main_primary]
}
```

There may be a minimum version requirement for using this feature, I'm not sure.

(For reference, `depends_on` is documented [here](https://www.terraform.io/docs/language/meta-arguments/depends_on.html).)

**Comment by satyamsareen** [comment]
> There is a workaround. You can take advantage of the fact that the first instance created in a cluster will be the writer, and subsequent instance(s) created will be read replica(s). Give the desired writer instance its own resource definition, and in the read replica instance resource definition(s) use `depends_on` to establish an explicit dependency on the writer instance. This will cause Terraform to create the writer instance first, making it the writer, and all others read replicas.

that's a clever way to do it @cbatson, thanks a ton 🎉 
Anyways, if we could get an option in the `aws_rds_cluster_instance` resource itself, it would be of much help

**Comment by Alexander-Giaquinto** [comment]
> > There is a workaround. You can take advantage of the fact that the first instance created in a cluster will be the writer, and subsequent instance(s) created will be read replica(s). Give the desired writer instance its own resource definition, and in the read replica instance resource definition(s) use `depends_on` to establish an explicit dependency on the writer instance. This will cause Terraform to create the writer instance first, making it the writer, and all others read replicas.
> 
> that's a clever way to do it @cbatson, thanks a ton tada Anyways, if we could get an option in the `aws_rds_cluster_instance` resource itself, something like `role="writer"` and `role="reader"` it would be of much help

This is exactly what the aws_rds_cluster_instance resource needs, but in the meantime has a workaround been discovered? I haven't ran my code yet but I am creating a cluster with two instances, if I run it what is going to be the role of the two instances?

**Comment by vijay-yadav-3** [comment]
It is open from 3 years, Cant believe this simple functionality is not present.
